components/TokenView/src/TokenView/TokenView.Events.cs

```diff
@@ -0,0 +1,160 @@
+    private void ScrollTabForwardButton_Click(object sender, RoutedEventArgs e)
+    {
+        if (_tokenViewScroller != null)
+        {
+            _tokenViewScroller.ChangeView(_tokenViewScroller.HorizontalOffset + _tokenViewScroller.ViewportWidth, null, null);
+        }
+    }
+
+    private void _tokenViewScroller_ViewChanging(object? sender, ScrollViewerViewChangingEventArgs e)
+    {
+#if !HAS_UNO
+        if (_tokenViewScrollBackButton != null)
+        {
+            if (e.FinalView.HorizontalOffset < 1)
+            {
+                _tokenViewScrollBackButton.Visibility = Visibility.Collapsed;
+            }
+            else if (e.FinalView.HorizontalOffset > 1)
+            {
+                _tokenViewScrollBackButton.Visibility = Visibility.Visible;
+            }
+        }
+
+        if (_tokenViewScrollForwardButton != null)
+        {
+            if (_tokenViewScroller != null)
+            {
+                if (e.FinalView.HorizontalOffset > _tokenViewScroller.ScrollableWidth - 1)
+                {
+                    _tokenViewScrollForwardButton.Visibility = Visibility.Collapsed;
+                }
+                else if (e.FinalView.HorizontalOffset < _tokenViewScroller.ScrollableWidth - 1)
+                {
+                    _tokenViewScrollForwardButton.Visibility = Visibility.Visible;
+                }
+            }
+        }
+#endif
+    }
+
+    private void ScrollViewer_Loaded(object sender, RoutedEventArgs e)
+    {
+        if (_tokenViewScroller != null)
+        {
+            _tokenViewScroller.Loaded -= ScrollViewer_Loaded;
+        }
+        if (_tokenViewScrollBackButton != null)
+        {
+            _tokenViewScrollBackButton.Click -= ScrollTabBackButton_Click;
+        }
+
+        if (_tokenViewScrollForwardButton != null)
+        {
+            _tokenViewScrollForwardButton.Click -= ScrollTabForwardButton_Click;
+        }
+
+        if (_tokenViewScroller != null)
+        {
+            _tokenViewScroller.ViewChanging += _tokenViewScroller_ViewChanging;
+            _tokenViewScrollBackButton = _tokenViewScroller.FindDescendant(TokenViewScrollBackButtonName) as ButtonBase;
+            _tokenViewScrollForwardButton = _tokenViewScroller.FindDescendant(TokenViewScrollForwardButtonName) as ButtonBase;
+        }
+
+        if (_tokenViewScrollBackButton != null)
+        {
+            _tokenViewScrollBackButton.Click += ScrollTabBackButton_Click;
+        }
+
+        if (_tokenViewScrollForwardButton != null)
+        {
+            _tokenViewScrollForwardButton.Click += ScrollTabForwardButton_Click;
+        }
+
+        UpdateScrollButtonsVisibility();
+    }
+
+    private void Token_Removing(object? sender, TokenItemRemovingEventArgs e)
+    {
+        if (ItemFromContainer(e.TokenItem) is object item)
+        {
+            var args = new TokenItemRemovingEventArgs(item, e.TokenItem);
+            TokenItemRemoving?.Invoke(this, args);
+
+            if (ItemsSource != null)
```

Comment:
Do we want to provide a hook to allow for the `Handled` pattern and prevent a token from being removed based on end-dev's code? We did this in the TokenizingTextBox with the deferred events (so they could use async with a deferral to pop-up a dialog for instance).

Maybe an issue we should log for future work? (Especially once we've ported those helpers to be in the common namespace.)